### ISSUE TITLE
Prisma Find Unique on Binary Columns

### ISSUE THREAD

**Issue by ati-pave** [prisma/prisma#20512]
### Bug description

Prisma findUnique on a unique binary column returns different result than Prisma findFirst on that column

### How to reproduce

Execute a `findUnique` query on a Binary Column from multiple threads that will get auto-batched by Prisma. 

Changing the `findUnique` -> `findFirst` fixes the problem, but both should return the same result

### Expected behavior

I expect `user1 === user2` in the below code snippet

### Prisma information
```
// prisma
model User {
   id Bytes @db.Binary(16) @id

  firstName String @db.VarChar(255)
 // .. other fields
}
```

```
// ts
const someParallelyCalledFunction = (id: Buffer[]) => {
    const user1 = user.findUnique({where: { id: id }, select: {id: true} })
    const user2 = user.findFirst({where: {id: id }, select: {id: true} })
}
```

In this case, user1 is `null` but user2 is not `null`. Both user1 and user2 should be equal

### Environment & setup

- OS: MacOS
- Database: MySQL
- Node.js version: v18.13.0

### Prisma Version

```
4.16.2
```

**Comment by aqrln** [comment]
I can reproduce this with Prisma 4.16.2:

```
prisma:query SELECT `repro-current`.`User`.`id` FROM `repro-current`.`User` WHERE `repro-current`.`User`.`id` IN (?,?)
{ id: <Buffer 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61> }
{ id: <Buffer 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61> }
prisma:query SELECT `repro-current`.`User`.`id` FROM `repro-current`.`User` WHERE `repro-current`.`User`.`id` = ? LIMIT ? OFFSET ?
{ id: <Buffer 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61> }
prisma:query SELECT `repro-current`.`User`.`id` FROM `repro-current`.`User` WHERE `repro-current`.`User`.`id` = ? LIMIT ? OFFSET ?
{ id: <Buffer 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61> }
```

However this is not reproducible anymore with recent Prisma versions:

```
prisma:query SELECT `repro-current`.`User`.`id` FROM `repro-current`.`User` WHERE `repro-current`.`User`.`id` IN (?,?)
findUnique { id: <Buffer 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61> }
findUnique { id: <Buffer 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61> }
prisma:query SELECT `repro-current`.`User`.`id` FROM `repro-current`.`User` WHERE `repro-current`.`User`.`id` = ? LIMIT ? OFFSET ?
findFirst { id: <Buffer 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61> }
prisma:query SELECT `repro-current`.`User`.`id` FROM `repro-current`.`User` WHERE `repro-current`.`User`.`id` = ? LIMIT ? OFFSET ?
findFirst { id: <Buffer 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61> }
```

`git bisect` points at 3a95e6ce0974a5df4ff5976eb11e79cef189e078, so this is a bug in the old protocol.

Please update to the latest Prisma version as this is already fixed.

If for whatever reason you can't update to Prisma 5 yet, alternatively you can enable the `jsonProtocol` preview feature in your current version instead, but updating to the latest version is highly recommended.

**Comment by ati-pave** [comment]
Thanks Alexey! We're blocked on updating to Prisma v5 (or enabling the `jsonProtocol`) due to another issue with Binary columns getting auto-encoded as Base64 strings (on write but not on read), but i think that has to do with our own usage of Prisma JSON columns.

I'll keep this in mind, thanks for the quick look into this!
